File: baek_1946.cpp
#include <stdio.h>
#include <algorithm>
#include <vector>

using namespace std;

int main(void) {
	int t, n, a, b, answer, tmp;
	vector<pair<int,int>> arr;
	scanf("%d", &t);
	for(int tcase=0; tcase<t; tcase++){
		arr.clear();
		answer = 1;
		scanf("%d", &n);
		for(int i=0; i<n; i++){
			scanf("%d %d", &a, &b);
			arr.push_back(make_pair(a,b));
		}
		sort(arr.begin(), arr.end());
		tmp = arr[0].second;
		for(int i=1; i<n; i++){
			if(arr[i].second<tmp){
				tmp = arr[i].second;
				answer++;
				if(tmp==1)
					break;
			}
		}
		printf("%d\n", answer);
	}
    return 0;
}

File: baek_1946.java
package com.company;

import java.util.*;

public class Main {
    static class Score implements Comparable<Score>{
        int first, second;
        Score(int first, int second){
            this.first = first;
            this.second = second;
        }

        @Override
        public int compareTo(Score score) {
            return Integer.compare(this.first, score.first);
        }
    }
    public static void main(String[] args) throws Exception {
        Scanner sc = new Scanner(System.in);
        List<Score> list = new ArrayList<>();
        int n, t, a, b, ans, tmp;
        t = sc.nextInt();
        for(int tcase=0; tcase<t; tcase++){
            list.clear();
            n = sc.nextInt();
            for(int i=0; i<n; i++){
                a = sc.nextInt();
                b = sc.nextInt();
                list.add(new Score(a,b));
            }
            Collections.sort(list);
            ans = 1;
            tmp = list.get(0).second;
            for(int i=1; i<n; i++){
                if(tmp>list.get(i).second){
                    tmp = list.get(i).second;
                    ans++;
                    if(tmp==1)
                        break;
                }
            }
            System.out.println(ans);
        }
    }
}
